src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(items:Item[], key:keyof Item):GroupsMap<Item> {
+  const obj:GroupsMap<Item> = {};
+
+  items.forEach((item) => {
+    const keyValue = String(item[key]);
+
+    if (!obj[keyValue]) {
+      obj[String(item[key])] = items.filter((el) => el[key] === item[key]);
+    }
```

Comment:
How about rewriting it this way?)

```suggestion
    if (!obj[keyValue]) {
      obj[keyValue] = [item];
    } else {
      obj[keyValue].push(item);
    }
```

Reply:
agree with Dima. This line is quite complicated to read